lib/matplotlib/scale.py

```diff
@@ -111,31 +93,38 @@ class LogTransformBase(Transform):
     def __init__(self, nonpos):
         Transform.__init__(self)
         if nonpos == 'mask':
-            self._handle_nonpos = _mask_non_positives
+            self._fill_value = np.nan
         else:
-            self._handle_nonpos = _clip_non_positives
-
-
-class Log10Transform(LogTransformBase):
-    base = 10.0
+            self._fill_value = 1e-300
 
     def transform_non_affine(self, a):
-        a = self._handle_nonpos(a * 10.0)
-        return np.log10(a)
-
-    def inverted(self):
-        return InvertedLog10Transform()
+        with np.errstate(invalid="ignore"):
+            a = np.where(a <= 0, self._fill_value, a)
+        np.log(a, a)
```

Comment:
Recommendation: `np.log(a, out=a)` keeps the memory and speed efficiency and improves the readability.